src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -1085,17 +1085,27 @@ bool clang_cpp_convertert::get_function_body(
           {
             exprt lhs;
             // parsing non-static member initializer
-            if (get_decl_ref(*init->getMember(), lhs))
+            if (get_decl_ref(*init->getMember(), lhs, false))
               return true;
 
-            build_member_from_component(
-              fd, lhs.id() == "dereference" ? lhs.op0() : lhs);
+            build_member_from_component(fd, lhs);
 
             exprt rhs;
             rhs.set("#member_init", 1);
             if (get_expr(*init->getInit(), rhs))
               return true;
 
+            if (lhs.type().reference())
+            {
+              /* Insert a cast such that (psuedo code) `int x = 1; int &y = x` is converted to
+               `int x = 1; int &y = &x`.
+               (for _actual_ variables, this is done in `get_var`) */
+              gen_typecast(ns, rhs, lhs.type());
+              /* Remove the reference from the type, otherwise `convert_reference`
+               in `clang_cpp_adjust` will try to dereference it, which is wrong. */
+              lhs.type().set("#reference", false);
```

Comment:
I'm not sure it's the best idea to remove the reference info here. Would it be possible to adapt convert_reference() (or rather adjust_reference()) such that for "assign" expressions to references, which are initializations, the dereferencing is not done?

Reply:
I agree, can we also move this handling to cpp_adjust? How does cbmc handles this code?

Reply:
Is https://github.com/esbmc/esbmc/pull/1802/commits/bd332b490f2947d281e740bcbb746dcf0b50bba0 more in line with what you expected?

Reply:
Or should the code rather be moved to https://github.com/esbmc/esbmc/blob/d564ede96e67d1fe073cd0a2dcc074539e5075fc/src/clang-c-frontend/clang_c_adjust_expr.cpp#L226-L230 ?
It is not always clear to me what belongs where if I'm honest.